### PR TITLE
fix the wandb logging issue

### LINKED ISSUE #33320: can't resume lora training due to wandb logging num params
### System Info

Hi,
I have some trained checkpoints that i'd like to resume from 
all of them are lora checkpoints 
but when resuming i get the following error in trainer

```
 trainer.train(resume_from_checkpoint=script_args.resume_from_checkpoint)
  File "/opt/conda/lib/python3.10/site-packages/transformers/trainer.py", line 1938, in train
    return inner_training_loop(
  File "/opt/conda/lib/python3.10/site-packages/transformers/trainer.py", line 2202, in _inner_training_loop
    self.control = self.callback_handler.on_train_begin(args, self.state, self.control)
  File "/opt/conda/lib/python3.10/site-packages/transformers/trainer_callback.py", line 460, in on_train_begin
    return self.call_event("on_train_begin", args, state, control)
  File "/opt/conda/lib/python3.10/site-packages/transformers/trainer_callback.py", line 507, in call_event
    result = getattr(callback, event)(
  File "/opt/conda/lib/python3.10/site-packages/transformers/integrations/integration_utils.py", line 900, in on_train_begin
    self.setup(args, state, model, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/transformers/integrations/integration_utils.py", line 853, in setup
    self._wandb.config["model/num_parameters"] = model.num_parameters()
  File "/root/.local/lib/python3.10/site-packages/wandb/sdk/wandb_config.py", line 149, in __setitem__
    key, val = self._sanitize(key, val)
  File "/root/.local/lib/python3.10/site-packages/wandb/sdk/wandb_config.py", line 258, in _sanitize
    raise config_util.ConfigError(
wandb.sdk.lib.config_util.ConfigError: Attempted to change value of key "model/num_parameters" from 0 to 266240
```
I assume that the fact that this is a lora training is relevant because the error describes a change in number or params (which shouldn't be logged as 0 from the first place)

and even though in line [integration_util.py#L838](https://github.com/huggingface/transformers/blob/43df47d8e78238021a4273746fc469336f948314/src/transformers/integrations/integration_utils.py#L838)

the wandb config dict was set with allow_val_change=True
i still get the above error. in line [/integration_utils.py#L853](https://github.com/huggingface/transformers/blob/43df47d8e78238021a4273746fc469336f948314/src/transformers/integrations/integration_utils.py#L853)

any idea on how to solve this?

Thanks

### Who can help?

@muellerzr @SunMarc

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

step 1. train a small model with dpo lora
step 2. try to resume with `trainer.train(resume_from_checkpoint=True)`
while setting 

```
 os.environ["WANDB_RESUME"] = "allow"
os.environ["WANDB_RUN_ID"] = script_args.run_id # same run_id as previous run
DPOConfig(
    output_dir=script_args.output_dir # same previous run out dir
    run_name=script_args.run_name # same run_name as previous run 
..
)

...

trainer.train(resume_from_checkpoint=True)
```

### Expected behavior

being able to resume previous training

### PR DESCRIPTION
**Pull request by ZIYU-DEEP**
# What does this PR do?

Fixes #33320. 

The existing error handling for wandb misses `ConfigError` when attempting to change the number of parameters, thus still blocking training; just added `ConfigError` to fix.

### CODE REVIEW

**Comment by vasqu** [comment]
Duplicate of #33440?

**Comment by ZIYU-DEEP** [comment]
ooooh same issue. the implementation difference is that `ConfigError` does not depend on `self` (cf., https://github.com/huggingface/transformers/pull/33440#discussion_r1759049149).

**Comment by vasqu** [comment]
Is it guaranteed that we have access to the wandb pkg in the file? If so, I'd also appreciate your variant of doing it by top-level import. Maybe you can help out with the other PR so that we can concentrate our focus on one :)

**Comment by vasqu** [comment]
Ohh whoops, ig the import can't be seen in `__init__` for the subsequent use (at least by the linter used here).

Guess it has to be moved into `setup` after we guarantee the successful import of wandb, i.e. https://github.com/huggingface/transformers/blob/bbef91ac735c6694a37b01e913f91fbd7a0b87f4/src/transformers/integrations/integration_utils.py#L804-L806
So based on `self._initialized` / after passing the first check that wandb is not None, it would be safe to import then.

**Review by LysandreJik — APPROVED** [review]
Thanks!

**Comment by LysandreJik** [comment]
There are a few code quality issues as @vasqu mentions above and can be seen in the code quality check; please fix those and I'll merge the PR.

**Comment by pengwei715** [comment]
Hi @ZIYU-DEEP Could you remove the space at the very beginning of line 809 of this file `src/transformers/integrations/integration_utils.py:809`, It should solve the issue.

File: src/transformers/integrations/integration_utils.py [hunk 1]
```diff
@@ -803,6 +803,10 @@ def setup(self, args, state, model, **kwargs):
         if self._wandb is None:
             return
         self._initialized = True
+
+        # prepare to handle potential configuration issues during setup
+        from wandb.sdk.lib.config_util import ConfigError as WandbConfigError
+
```
Comment (pengwei715) [inline src/transformers/integrations/integration_utils.py:809]: Hi @ZIYU-DEEP Could you remove the space at the very beginning of line 809 of this file src/transformers/integrations/integration_utils.py:809, It should solve the issue.

Comment (ZIYU-DEEP) [reply src/transformers/integrations/integration_utils.py]: thanks so much @pengwei715 for catching this!

**Review by pengwei715 — APPROVED** [review]

**Review by LysandreJik — APPROVED** [review]
Thanks!

**Pull request merged**